bin/tx-prover/src/proxy/metrics.rs

```diff
@@ -0,0 +1,81 @@
+use std::sync::LazyLock;
+
+use prometheus::{
+    register_histogram, register_int_counter, register_int_counter_vec, register_int_gauge,
+    Histogram, IntCounter, IntCounterVec, IntGauge,
+};
+
+// QUEUE METRICS
+// ================================================================================================
+
+pub static QUEUE_SIZE: LazyLock<IntGauge> =
+    LazyLock::new(|| register_int_gauge!("queue_size", "Number of requests in the queue").unwrap());
+pub static QUEUE_LATENCY: LazyLock<Histogram> = LazyLock::new(|| {
+    register_histogram!(
+        "queue_latency",
+        "Time requests spend in the queue",
+        vec![0.1, 0.5, 1.0, 2.0, 5.0, 10.0]
+    )
+    .unwrap()
+});
+pub static QUEUE_DROP_COUNT: LazyLock<IntCounter> = LazyLock::new(|| {
+    register_int_counter!("queue_drop_count", "Number of requests dropped due to a full queue")
+        .unwrap()
+});
+
+// WORKER METRICS
+// ================================================================================================
+
+pub static WORKER_COUNT: LazyLock<IntGauge> =
+    LazyLock::new(|| register_int_gauge!("worker_count", "Number of workers").unwrap());
+pub static WORKER_UNHEALTHY: LazyLock<IntCounter> = LazyLock::new(|| {
+    register_int_counter!("worker_unhealthy", "Number of unhealthy workers").unwrap()
+});
+pub static WORKER_UTILIZATION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!("worker_utilization", "Number of requests being processed by workers")
+        .unwrap()
+});
```

Comment:
Is the description here correct? I thought this metric tracks the number of busy workers.

Reply:
It's not quite correct. This represents the number of workers handling requests. I'm updating the description.

Reply:
Also I'm renaming the metric to `WORKER_BUSY`